csp-mixin/signals/virtualMachinesOverview.libsonnet

```
@@ -0,0 +1,181 @@
+            legendCustomTemplate: '',
+          },
+        },
+      },
+
+      top5DiskWrite: {
+        name: 'Top 5 Instances - Disk write bytes',
+        description: 'List of top 5 Instances by disk write bytes',
+        type: 'raw',
+        unit: 'decbytes',
+        sources: {
+          azuremonitor: {
+            expr: 'topk(5, sum by (resourceName, job, resourceGroup, subscriptionName)(azure_microsoft_compute_virtualmachines_disk_write_bytes_total_bytes{%(queriesSelector)s}))',
+            legendCustomTemplate: '',
+          },
+        },
+      },
+
+      diskReadBytes: {
+        name: 'Disk bytes (total)',
+        description: 'Bytes read/written from/to disk during monitoring period',
+        type: 'raw',
+        unit: 'decbytes',
+        sources: {
+          azuremonitor: {
+            expr: 'sum(rate(azure_microsoft_compute_virtualmachines_disk_read_bytes_total_bytes{%(queriesSelector)s}[$__rate_interval]))',
+            legendCustomTemplate: 'Read',
+          },
+        },
+      },
+
+      diskWriteBytes: {
+        name: 'Disk write bytes',
+        description: 'Bytes written to disk during monitoring period',
+        type: 'raw',
+        unit: 'decbytes',
+        sources: {
+          azuremonitor: {
+            expr: 'sum(rate(azure_microsoft_compute_virtualmachines_disk_write_bytes_total_bytes{%(queriesSelector)s}[$__rate_interval]))',
+            legendCustomTemplate: 'Write',
+          },
+        },
+      },
+
+      diskReadOperations: {
+        name: 'Disk Operations/Sec (average)',
+        description: 'Disk Read/Write IOPS',
+        type: 'raw',
+        unit: 'cps',
+        sources: {
+          azuremonitor: {
+            expr: 'sum(rate(azure_microsoft_compute_virtualmachines_disk_read_operations_sec_average_countpersecond{%(queriesSelector)s}[$__rate_interval]))',
+            legendCustomTemplate: 'Read',
+          },
+        },
+      },
+
+      diskWriteOperations: {
+        name: 'Disk Write Operations',
+        description: 'Disk Write IOPS',
+        type: 'raw',
+        unit: 'cps',
+        sources: {
+          azuremonitor: {
+            expr: 'sum(rate(azure_microsoft_compute_virtualmachines_disk_write_operations_sec_average_countpersecond{%(queriesSelector)s}[$__rate_interval]))',
+            legendCustomTemplate: 'Write',
+          },
+        },
+      },
+
+      networkInTotal: {
+        name: 'Network throughput send/received',
+        description: 'The number of bytes sent/received over the network.',
+        type: 'raw',
+        unit: 'decbytes',
+        sources: {
+          azuremonitor: {
+            expr: 'sum(rate(azure_microsoft_compute_virtualmachines_network_in_total_total_bytes{%(queriesSelector)s}[$__rate_interval]))',
+            legendCustomTemplate: 'Received',
+          },
+        },
+      },
```

just another example how to turn this to type=counter form:
```suggestion
      networkInTotal: {
        name: 'Network throughput send/received',
        description: 'The number of bytes sent/received over the network.',
        type: 'counter',
        unit: 'decbytes',
        sources: {
          azuremonitor: {
            expr: 'azure_microsoft_compute_virtualmachines_network_in_total_total_bytes{%(queriesSelector)s},
            exprWrappers: ['sum(',')'], 
            legendCustomTemplate: 'Received',
          },
        },
      },
```

// by default it will aggregate by aggLevel=instance, so we can add additional wrapper applied at the end, if we want to aggregate even further than instance or group level.

Keeping the expr form simplier would allow also use those expressions inside alerts, by using signal.x.asRuleExpression() function, (where [5m] aggregation would be applied instead of grafana-only $__rate_interval.